tutorials/Creating a new Lumigator endpoint.md

```diff
@@ -0,0 +1,581 @@
+
+Note how similar this is to some of the other services (e.g. `DatasetService`): you can expect this from services which
+only deal with the DB as the main operations you'll do are those who operate on a table (create, delete, get, list,
+etc). You will likely see a different behavior in more advanced endpoints (e.g. those which involve running ray jobs),
+but we'll discuss that in another tutorial.
+
+#### 2.4. Inject dependencies into a TaskService
+
+As `TaskService` depends on the existence of a database, we should inject a dependency on a DB session. To do this, add
+the following code to 
+[`backend/api/deps.py`](https://github.com/mozilla-ai/lumigator/blob/main/lumigator/python/mzai/backend/api/deps.py):
+
+```
+def get_task_service(session: DBSessionDep) -> TaskService:
+    task_repo = TaskRepository(session)
+    return TaskService(task_repo)
+
+
+TaskServiceDep = Annotated[TaskService, Depends(get_task_service)]
+```
+
+#### 2.5. Update routes
+
+You are almost there! The last thing you need to do is update the `backend/api/routes/tasks.py`file with new code which
+will map API requests to `TaskService` methods:
+
+```
+from uuid import UUID
+
+from fastapi import APIRouter, status
+
+from mzai.backend.api.deps import TaskServiceDep
+from mzai.schemas.extras import ListingResponse
+from mzai.schemas.tasks import TaskCreate, TaskResponse
+
+router = APIRouter()
+
+
+@router.post("/", status_code=status.HTTP_201_CREATED)
+def create_task(service: TaskServiceDep, request: TaskCreate) -> TaskResponse:
+    return service.create_task(request)
+
+
+@router.get("/{task_id}")
+def get_task(service: TaskServiceDep, task_id: UUID) -> TaskResponse:
+    return service.get_task(task_id)
+
+
+@router.delete("/{task_id}", status_code=status.HTTP_204_NO_CONTENT)
+def delete_task(service: TaskServiceDep, task_id: UUID) -> None:
+    service.delete_task(task_id)
+
+
+@router.get("/")
+def list_tasks(
+    service: TaskServiceDep,
+    skip: int = 0,
+    limit: int = 100,
+) -> ListingResponse[TaskResponse]:
+    return service.list_tasks(skip, limit)
+```
+
+#### 2.6. Test
+
+To test your new endpoint, connect to [http://localhost/docs](http://localhost/docs). You should see a new section like
+the following one:
+
+<p style="text-align: center;">
+<img src="assets/image-20240913-083603.png" alt="List of new methods in the tasks endpoint" >
+List of new methods in the tasks endpoint
+</p>
+
+Here is an example for the creation of a new task:
+
+<p style="text-align: center;">
+<img src="assets/image-20240913-083904.png" alt="New task creation" >
+New task creation
+</p>
+
+And here is the format of a task list (to which we have also added a `summarization_eval` task):
+
+<p style="text-align: center;">
+<img src="assets/image-20240913-084103.png" alt="The current list of tasks on the system" >
+The current list of tasks on the system
+</p>
+
+
+## Conclusions
```

Comment:
I don't think we need a conclusion section for a README

Reply:
To my defense I thought this was a tutorial when I started :-) 
I kept the contents (as I'd like to wrap things up and provide a link to the actual code) but changed the title to "final checks and reference code"